lib/NodeInstantiator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\RST;
+
+use Doctrine\RST\Nodes\Node;
+use InvalidArgumentException;
+use ReflectionClass;
+use function sprintf;
+
+class NodeInstantiator
```

Comment:
shouldn't it be wise to create an interface for this ? are we sure the `create()` function here will fit every needs ?

Reply:
I wasn't intending for this to be extended, just used. You can use `DefaultNodeFactory` or implement your own instance of the `NodeFactory` interface. The`DefaultNodeFactory` just let's you configure and customize the class to use for each node type without having to implement your own `NodeFactory`.